### PR TITLE
My hw 04

### PR DESCRIPTION
**Pull request by tsekhanskaya**

### CODE REVIEW

**Comment by vladhilko** [comment]
CI is red 😞

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+require 'time'
+
+class Mentor
+  atter_accessor :name, :surname, :noties_for_student, :hw
+  def initialize
+    @name = name
+    @surname = surname
+    @noties_for_student = []
+    @list_students = []
+    @hw = []
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:12]: Could you please write the whole word?

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  # mentor see notification about homework to work
+  # mentor see notification about homework to check
+  def notifications
+    puts noties_for_mentors
+  end
+
+  # mentor mark as read all notifications
+  def mark_as_read
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:38]: The goal of the method is to make the notifications' storage empty

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  # mentor subscribe to student
+  def subscribe_to
+    list_students << Student.full_name
+  end
+
+  # mentor see notification about homework to work
+  # mentor see notification about homework to check
+  def notifications
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:33]: There's no need in this method.

File: HW04/anna_tsekhanskaya/lib/student.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                                          description: "Student #{Student.full_name} mark as read at #{Time.now}")
+  end
+
+  # student can see homeworks
+  def homeworks
+    Homework.Mentor.see_homework
+  end
+
+  # student take to work homework [1]
+  def to_work(homework)
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/student.rb:37]: If a student takes a homework to work he should add the homework to his homeworks' list

File: HW04/anna_tsekhanskaya/lib/mentor.rb (hunk 3 above)
Comment (DashaSledopit) [reply HW04/anna_tsekhanskaya/lib/mentor.rb:34]: Once again :) There's no need in this method at all. To make `mentor.notifications` and `student.notifications` work you need to change the name of `noties_for_student` & `noties_for_mentor`  and a little of its usage.

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    noties_for_student.clear
+    Notification.mark_as_read = true
+  end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:37]:
```suggestion
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'time'
+
+class Mentor
+  atter_accessor :name, :surname, :noties_for_student, :homework_answer
+  def initialize
+    @name = name
+    @surname = surname
+    @noties_for_student = []
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:10]:
```suggestion
    @notifications = []
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 7]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'time'
+
+class Mentor
+  atter_accessor :name, :surname, :noties_for_student, :homework_answer
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:6]:
```suggestion
  atter_accessor :name, :surname, :notifications, :homework_answer, :student
```

Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:6]:
```suggestion
  attr_reader :name, :surname, :notifications, :homeworks
  attr_accessor :student
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  def initialize
+    @name = name
+    @surname = surname
+    @noties_for_student = []
+    @list_students = []
+    @homework_answer = []
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:12]:
```suggestion
    @homeworks = []
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+  # mentor reject homework
+  def reject_to_work(_homework)
+    noties_for_student << Notification.new(title: 'Reject',
+                                           description: "Mentor #{Mentor.full_name} reject homework at #{Time.now}")
+  end
+
+  # mentor accept homework
+  def accept
+    noties_for_student << Notification.new(title: 'Accept homework',
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:52]:
```suggestion
    student.notifications << Notification.new(title: 'Accept homework',
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  def subscribe_to
+    list_students << Student.full_name
+  end
+
+  # mentor see notification about homework to work
+  # mentor see notification about homework to check
+  # def notifications
+  #  notifications.clear
+  #  Notification.mark_as_read = true
+  # end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:40]:
```suggestion

```

File: HW04/anna_tsekhanskaya/lib/student.rb [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  def full_name
+    "#{name.capitalize} #{surname.capitalize}"
+  end
+
+  # student see notification about new homework
+  # student see notification about homework was reject
+  # student see notification about homework was accept
+  # def notifications
+  # notifications.clear
+  # end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/student.rb:26]:
```suggestion

```

File: HW04/anna_tsekhanskaya/lib/homework.rb [hunk 12]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Homework
+  def initialize
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/homework.rb:4]:
```suggestion
  attr_reader :student, :mentor, :title

  def initialize
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 13]
```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'time'
+
+class Mentor
+  attr_reader :name, :surname, :notifications, :homeworks
+  attr_accessor :student
+
+  def initialize
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:9]: attr_reader :homeworks, :name, :surname, :list_students, :notifications
  attr_accessor :student

  def initialize(name:  , surname: )

Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:9]:
```suggestion
attr_reader :homeworks, :name, :surname, :list_students, :notifications
attr_accessor :student

def initialize(name: , surname: )
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  # mentor add new homework
+  def add_homework(title, description, student)
+    student.notifications << Notification.new(title: title,
+                                              description: "#{mentor.full_name} added homework at #{Time.now}",
+                                              student: student.full_name)
+    homeworks << Homework.new(title: title, description: description, student: Student.full_name,
+                              mentor: Mentor.full_name)
+  end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:28]:
```suggestion
  def add_homework(title: , description: , student: )
    homeworks << Homework.new(title: title, description: description, student: student.full_name,
                              mentor: self.full_name)
    student.notifications << Notification.new(title: title,
                                              description: "#{self.full_name} added homework at #{Time.now}")
  end
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+  # mentor subscribe to student
+  def subscribe_to
+    list_students << Student.full_name
+  end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:33]:
```suggestion
  def subscribe_to!(student)
    list_students << student
  end
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+  # def notifications
+  #  notifications.clear
+  #  Notification.mark_as_read = true
+  # end
+
+  # mentor mark as read all notifications
+  def mark_as_read
+    Notification.mark_as_read = true
+    student.notifications.clear
+  end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:46]:
```suggestion
  def mark_as_read!
    notifications.clear
  end
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+  # mentor reject homework
+  def reject_to_work(_homework)
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:49]:
```suggestion
  def reject_to_work!(homework)
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    student.notifications << Notification.new(title: 'Reject',
+                                              description: "Mentor #{Mentor.full_name} reject homework at #{Time.now}")
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:51]:
```suggestion
                                              description: "Mentor #{self.full_name} reject homework at #{Time.now}")
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  # mentor accept homework
+  def accept
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:55]:
```suggestion
  def accept!(homework)
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+    student.notifications << Notification.new(title: 'Accept homework',
+                                              description: "Mentor #{Mentor.full_name} accept homework at #{Time.now}")
+  end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:58]:
```suggestion
  def accept!(homework)
    notifications << Notification.new(title: 'Accept homework',
                                              description: "Mentor #{Mentor.full_name} accept homework at #{Time.now}")
  end
```

File: HW04/anna_tsekhanskaya/lib/mentor.rb [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+  def mark_as_read
+    Notification.mark_as_read = true
+    student.notifications.clear
+  end
+
+  # mentor reject homework
+  def reject_to_work(_homework)
+    student.notifications << Notification.new(title: 'Reject',
+                                              description: "Mentor #{Mentor.full_name} reject homework at #{Time.now}")
+  end
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/mentor.rb:52]:
```suggestion
  def reject_to_work!(homework)
    notifications << Notification.new(title: 'Reject',
                                              description: "Mentor #{self.full_name} reject homework at #{Time.now}")
  end
```

File: HW04/anna_tsekhanskaya/lib/notification.rb [hunk 22]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+require 'time'
+
+class Notification
+  atter_accessor :title, :description
+  def initialize
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/notification.rb:7]:
```suggestion
  atter_accessor :title, :description
 
  def initialize(title: , description: )
```

File: HW04/anna_tsekhanskaya/lib/notification.rb [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
+    @title = title
+    @description = description
+    @mark_as_read = false
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/notification.rb:10]:
```suggestion
```

File: HW04/anna_tsekhanskaya/lib/student.rb [hunk 24]
```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'time'
+
+class Student
```
Comment (DashaSledopit) [inline HW04/anna_tsekhanskaya/lib/student.rb:5]: Please, look at the comments to the Mentor class and try to correct this one accordingly

**Comment by B0RlS** [comment]
Merged but failed due to deadlines

**Pull request merged**